### PR TITLE
Update to Mimikatz 2.1.1 20170608 for changentlm function

### PR DESCRIPTION
**Pull request by OJ**
Associated Meterpreter PR here: https://github.com/rapid7/metasploit-payloads/pull/204

This PR includes support for @gentilkiwi's new `changentlm` feature in `kiwi`. This feature allows you to change the NTLM password for a given account.

Relevant tweets:
* https://twitter.com/gentilkiwi/status/872588656967548928
* https://twitter.com/gentilkiwi/status/872590369820737541

This apparently avoids the `setpassword` event logging. It also means that an account password can be temporarily set to something known, and then set back to the original unknown thing once the account has been used. Genius.

## Sample run

image

## Verification

- [x] Create a Windows Meterpreter session with a native arch (eg. x64 on windows x64).
- [x] Use the new `kiwi` extension.
- [x] Make sure the `password_change` feature is active in the help menu.
- [x] Check the help options by running `password_change -h`
- [x] Validate that the password can be changed using the combination of parameters shown.
- [x] Specifying the `-op` and `-oh` flags should result in an error.
- [x] Specifying the `-np` and `-nh` flags should result in an error.
- [x] The `-u` flag should be mandatory.
- [x] The `-s` flag should be optional (defaulting to the localhost if not specified).
- [x] Useful messages should be shown on errors (invalid server, invalid user, etc).
- [x] The new NTLM hash should be rendered on success.

### CODE REVIEW

**Comment by OJ** [comment]
Note that this PR relies on landing #8523 first due to the fact that the Meterpreter payloads repo already has updated binaries for that PR in master.

**Comment by p3nt4** [comment]
Nice work. 

There is a post module with the same intent (that I wrote):
https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/manage/hashcarve.rb

I started working on the same idea with cached credentials but did not have the skills to make it work, there seems to be some checksums at the end of the cached entry that I could not bypass. Would be nice if someone more talented had a crack at it!

Do you know if this will ever be supported?

**Comment by busterb** [comment]
1.2.35 has the metasploit-framework update, would you mind rebasing on master and bumping paylods? Right now this points to a couple of outdated guid commits.

**Comment by OJ** [comment]
Roger, on it now!

**Comment by OJ** [comment]
OK doneski! Updated cached sizes too, hopefully all is good now. Thanks Brent!

**Comment by busterb** [comment]
Thanks!

**Comment by busterb** [comment]
## Release Notes

This updates the Windows Meterpreter kiwi extension with the latest version of mimikatz, which adds the ability to change Windows passwords without triggering event logging via the `password_change` command.

**Comment by OJ** [comment]
Thanks Brent!

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.4.1
+2.4.0
```
Comment (OJ) [inline .ruby-version]: My bad! Sorry. I built this on OSX and 2.4.1 is a nightmare hence the local change.

**Pull request merged**